### PR TITLE
DRIVERS-2711: Publish Topology Description Changed Event on Topology Close

### PR DESCRIPTION
**Pull request by W-A-James**
What's changing?

- load-balancer spec now explicitly requires that a `TopologyDescriptionChangedEvent` transitioning the topology from `LoadBalanced` to `Unknown` is emitted just before the `TopologyClosedEvent` is emitted.
- sdam monitoring spec new explicitly requires that a `TopologyDescriptionChangedEvent` transitioning the topology to `Unknown` is emitted just before the `TopologyClosedEvent` is emitted.
- load-balanced logging test asserts that `TopologyDescriptionChangedEvent` is emitted just before `TopologyClosedEvent`
- Add monitoring tests to check for new behaviour across all topology types

Please complete the following before merging:

- [x] Update changelog.
- [x] Make sure there are generated JSON files from the YAML test files.
- [x] Test changes in at least one language driver.
- [x] Test these changes against all server versions and topologies (including standalone, replica set, sharded
  clusters, and serverless).

### CODE REVIEW

**Review by ShaneHarvey — DISMISSED** [review]
Let's add SDAM monitoring tests for this change (that don't depend on logging).

File: source/server-discovery-and-monitoring/tests/unified/standalone-topology-lifecycle.json [hunk 1]
```diff
@@ -0,0 +1,78 @@
+{
```
Comment (ShaneHarvey) [inline source/server-discovery-and-monitoring/tests/unified/standalone-topology-lifecycle.json:1]: Should this file be removed? I notice there's no yaml verison.

Comment (W-A-James) [reply source/server-discovery-and-monitoring/tests/unified/standalone-topology-lifecycle.json]: Good catch. Deleted.

File: source/server-discovery-and-monitoring/server-discovery-and-monitoring-logging-and-monitoring.rst [hunk 2]
```diff
@@ -117,6 +117,12 @@ Initial Topology Description
 
 The first ``TopologyDescriptionChangedEvent`` to be emitted from a monitored Topology MUST set its ``previousDescription`` property to be a ``TopologyDescription`` object in the "unknown" state.
 
+Closing Topology Description
+----------------------------
+
+When a ``Topology`` object or equivalent is being shut-down or closed, the driver SHOULD change the
+``TopologyDescription`` to an "unknown" state.
```
Comment (alcaeus) [inline source/server-discovery-and-monitoring/server-discovery-and-monitoring-logging-and-monitoring.rst:124]: Nit: this conflicts with the requirement in the load balancer spec that there must be an event that transitions the topology to `unknown`, whereas this leaves it optional. I suggest changing this to `MUST` as well.
The "replicaset-emit-topology-description-changed-before-close" test confirms this by explicitly checking for an `unknown` state on the topology description.

**Review by alcaeus — CHANGES_REQUESTED** [review]

**Review by alcaeus — APPROVED** [review]

**Pull request merged**